main.py

```diff
@@ -111,6 +126,38 @@ def update_colors(self, links):
 
         self._send_flow_mods(dpid_flows)
 
+    def handle_link_disabled(self, link):
+        """Handle link deletion. Deletes only flows from the proper switches.
+         The field 'neighbors' is managed by update_colors method."""
+        switch_a_id = link.endpoint_a.switch.dpid
+        switch_b_id = link.endpoint_b.switch.dpid
+
+        if switch_b_id == switch_a_id:
+            return
+        
+        with self._switches_lock:
+            flow_mods = defaultdict(list)
+            flow = self.switches[switch_a_id]['flows'][switch_b_id]
+            flow_mods[switch_a_id].append({
+                "table_id": flow['table_id'],
+                "match": flow['match']
+            })
+            flow = self.switches[switch_b_id]['flows'][switch_a_id]
+            flow_mods[switch_b_id].append({
+                "table_id": flow['table_id'],
+                "match": flow['match']
+            })
+            self.switches[switch_a_id]['flows'].pop(switch_b_id)
+            self.switches[switch_b_id]['flows'].pop(switch_a_id)
+        self._remove_flow_mods(flow_mods)
+
+    def handle_switch_disabled(self, dpid):
+        """Handle switch deletion. Links are expected to be disabled first
+         therefore the deleted inner dictionary is expected to be empty with
```

Comment:
@Alopalao excellent docstring, and this probably won't change in the short-term/medium-term, but can you also add a conditional verifying if indeed it's empty, and if it's not to log as an error with the content (error instead of warning since we're expecting that this won't happen based on the existing guarantees)

Reply:
Great stuff.